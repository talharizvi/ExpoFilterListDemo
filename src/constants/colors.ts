export default {
    primary: '#ffffff',
    secondary: '#dadee6',
    selectedOptionColor:'#000000',
    background: 'rgba(0, 122, 255, 0.12)',
    iconText: '#5280eb',
    userText:'#5e5e5d',
    loader:'#007bff'
  };