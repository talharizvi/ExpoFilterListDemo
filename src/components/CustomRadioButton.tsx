import React, { useState, useCallback } from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import colors from '../constants/colors';

interface CustomRadioButtonProps {
    handleSelectedUserType:(userType:string)=>void
}

const CustomRadioButton:React.FC<CustomRadioButtonProps> = ({handleSelectedUserType}) => {
    const [selectedOption, setSelectedOption] = useState<string>('admin');
  
    const handleOptionSelect = useCallback((option: string) => {
      if (option === selectedOption) {
          return;
      }
      setSelectedOption(option);
      handleSelectedUserType(option);
  }, [selectedOption, handleSelectedUserType]);
  
    return (
      <View>
        <TouchableOpacity
          style={[styles.itemView, selectedOption === 'admin' && styles.selectedOptionBackground]}
          onPress={() => {
            handleOptionSelect('admin')
        }}
        >
          <View
            style={styles.childView}
          >
            {selectedOption === 'admin' && (
              <View style={styles.selectedOptionView} />
            )}
          </View>
          <Text style={styles.itemTextStyle}>Admin</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={[styles.itemView, selectedOption === 'manager' && styles.selectedOptionBackground]}
          onPress={() => {
            handleOptionSelect('manager')
        }}
        >
          <View
            style={styles.childView}
          >
            {selectedOption === 'manager' && (
              <View style={styles.selectedOptionView} />
            )}
          </View>
          <Text style={styles.itemTextStyle}>Manager</Text>
        </TouchableOpacity>
      </View>
    );
  };

const styles = StyleSheet.create({
    itemView:{
        flexDirection: 'row', 
        alignItems: 'center', 
        marginRight: 20, 
        marginTop: 10, 
        borderRadius:8, 
        paddingVertical:12, 
        paddingLeft:8
    },
    childView:{
        height: 18,
        width: 18,
        borderRadius: 12,
        borderWidth: 2,
        borderColor: colors.secondary,
        alignItems: 'center',
        justifyContent: 'center',
    },
    selectedOptionView:{
        height: 6, 
        width: 6, 
        borderRadius: 3, 
        backgroundColor: colors.selectedOptionColor
    },
    itemTextStyle:{ 
        marginLeft: 18 
    },
    selectedOptionBackground:{
        backgroundColor:colors.background
    }
})

export default CustomRadioButton;