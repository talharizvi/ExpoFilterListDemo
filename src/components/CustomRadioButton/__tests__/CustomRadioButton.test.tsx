import React from 'react';
import { fireEvent, render, screen, waitFor,  } from '@testing-library/react-native';
import CustomRadioButton from '../CustomRadioButton';
import colors from '../../../constants/colors';

describe('CustomRadioButton', () => {
  const handleSelectedUserType = jest.fn();

  it('renders the CustomRadioButton', () => {
    render(<CustomRadioButton handleSelectedUserType={handleSelectedUserType} />);
    expect(screen.getByText('Admin')).toBeDefined();
    expect(screen.queryByText('Manager')).toBeDefined();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('calls handleSelectedUserType with "admin" when default option is selected', () => {
    render(<CustomRadioButton handleSelectedUserType={handleSelectedUserType} />);
    expect(screen.getByTestId('admin-button').props.style.backgroundColor).toBe(colors.background);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('calls handleSelectedUserType with "manager" when "Manager" option is selected', () => {
    render(<CustomRadioButton handleSelectedUserType={handleSelectedUserType} />);
    fireEvent.press(screen.getByText('Manager'));
    expect(handleSelectedUserType).toHaveBeenCalledWith('manager');
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('calls handleSelectedUserType with "admin" when "admin" option is selected', () => {
    render(<CustomRadioButton handleSelectedUserType={handleSelectedUserType} />);
    fireEvent.press(screen.getByTestId('admin-button'));
    expect(screen.getByTestId('admin-button').props.style.backgroundColor).toBe(colors.background);
    expect(screen.toJSON()).toMatchSnapshot();
  });

});