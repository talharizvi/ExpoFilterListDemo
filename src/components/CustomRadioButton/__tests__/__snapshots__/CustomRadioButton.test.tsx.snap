// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRadioButton calls handleSelectedUserType with "admin" when "admin" option is selected 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 122, 255, 0.12)",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
        "paddingLeft": 8,
        "paddingVertical": 12,
      }
    }
    testID="admin-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#dadee6",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#000000",
            "borderRadius": 3,
            "height": 6,
            "width": 6,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "marginLeft": 18,
        }
      }
    >
      Admin
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
        "paddingLeft": 8,
        "paddingVertical": 12,
      }
    }
    testID="manager-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#dadee6",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    />
    <Text
      style={
        {
          "marginLeft": 18,
        }
      }
    >
      Manager
    </Text>
  </View>
</View>
`;

exports[`CustomRadioButton calls handleSelectedUserType with "admin" when default option is selected 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 122, 255, 0.12)",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
        "paddingLeft": 8,
        "paddingVertical": 12,
      }
    }
    testID="admin-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#dadee6",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#000000",
            "borderRadius": 3,
            "height": 6,
            "width": 6,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "marginLeft": 18,
        }
      }
    >
      Admin
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
        "paddingLeft": 8,
        "paddingVertical": 12,
      }
    }
    testID="manager-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#dadee6",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    />
    <Text
      style={
        {
          "marginLeft": 18,
        }
      }
    >
      Manager
    </Text>
  </View>
</View>
`;

exports[`CustomRadioButton calls handleSelectedUserType with "manager" when "Manager" option is selected 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
        "paddingLeft": 8,
        "paddingVertical": 12,
      }
    }
    testID="admin-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#dadee6",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    />
    <Text
      style={
        {
          "marginLeft": 18,
        }
      }
    >
      Admin
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 122, 255, 0.12)",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
        "paddingLeft": 8,
        "paddingVertical": 12,
      }
    }
    testID="manager-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#dadee6",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#000000",
            "borderRadius": 3,
            "height": 6,
            "width": 6,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "marginLeft": 18,
        }
      }
    >
      Manager
    </Text>
  </View>
</View>
`;

exports[`CustomRadioButton renders the CustomRadioButton 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 122, 255, 0.12)",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
        "paddingLeft": 8,
        "paddingVertical": 12,
      }
    }
    testID="admin-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#dadee6",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#000000",
            "borderRadius": 3,
            "height": 6,
            "width": 6,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "marginLeft": 18,
        }
      }
    >
      Admin
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginRight": 20,
        "marginTop": 10,
        "opacity": 1,
        "paddingLeft": 8,
        "paddingVertical": 12,
      }
    }
    testID="manager-button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#dadee6",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 18,
          "justifyContent": "center",
          "width": 18,
        }
      }
    />
    <Text
      style={
        {
          "marginLeft": 18,
        }
      }
    >
      Manager
    </Text>
  </View>
</View>
`;
